Enable cross build against scala 2.12 and 2.13

aumann opened the pull request:
Fixes #136

Adds htt4s-0-21 (http4s-0-20 only built upto 2.13_M5)

aumann commented on sbt-scalapact-nodeps/src/main/scala/com/itv/scalapact/plugin/ScalaPactPlugin.scala:
```diff
@@ -10,6 +10,8 @@ import sbt.plugins.JvmPlugin
 import sbt.{Def, _}
 import complete.DefaultParsers._
 
+import scala.language.implicitConversions
```
That's the only 'real' change here. The rest has is scalafmt

dantb commented on build.sbt:
```diff
@@ -331,7 +394,7 @@ lazy val plugin =
     .dependsOn(argonaut62)
     .dependsOn(http4s018)
-    .settings(compilerOptions212: _*)
+    .settings(scalacOptions ++= compilerOptionsVersion(scalaVersion.value))
```
Ditto but I guess this wants `scala212OnlySettings`?

dantb commented on build.sbt:
```diff
@@ -376,7 +440,7 @@ lazy val standalone =
     .dependsOn(circe09)
     .dependsOn(http4s018)
-    .settings(compilerOptions212: _*)
+    .settings(scalacOptions ++= compilerOptionsVersion(scalaVersion.value))
```
Ditto

aumann commented:
Sorry for the silence, will tackle the remaining issues today.

aumann commented on scalapact-http4s-0-20/src/main/scala/com/itv/scalapact/http4s20/impl/Http4sClientHelper.scala:
```diff
@@ -23,13 +23,17 @@ object Http4sClientHelper {
+    )(ctx =>
+      builder.withSslContext(ctx)
+    ).resource
+    builder.resource
```
oh, yes. Didn't know about the SSLContext.getDefault -- much better

aumann commented on build.sbt:
```diff
@@ -317,7 +380,7 @@ lazy val pluginShared =
     )
     .dependsOn(core)
-    .settings(compilerOptions212: _*)
+    .settings(scalacOptions ++= compilerOptionsVersion(scalaVersion.value))
```
I changed these for uniformity, but left them at 2.12 -- to be honest I am not sure about cross compilation in the sbt plugin stuff

aumann commented:
I addressed the comments w.r.t. the builders / build.sbt

Concerning the WartRemover.Any stuff:

- I tried fixing the http4s response warnings
- However, afterwards there were warnings everywhere in com.itv.scalapact.ScalaPactVerify
- Maybe this is related to https://github.com/wartremover/wartremover/issues/454

davesmith00000 commented:
Hi @aumann - sorry - our turn to be slow! Just wanted to say that this hasn't been forgotten, I'll try and take a look in the next couple of days.

vilunov commented:
@dantb @davesmith00000 Hi! Is there any progress on reviewing this PR?

orium commented:
Thanks @aumann.  I went ahead and solved the conflicts with master and did some minor tweaking.
